Examples/Scripts/Python/fatras.py

```diff
@@ -36,16 +37,22 @@ def addFatras(
     rnd = rnd or acts.examples.RandomNumbers()
 
     # Selector
-    selector = acts.examples.ParticleSelector(
-        level=s.config.logLevel,
-        inputParticles="particles_input",
-        outputParticles="particles_selected",
-    )
+    if preselectParticles:
+        particles_selected = "particles_selected"
+        s.addAlgorithm(
+            acts.examples.ParticleSelector(
+                level=s.config.logLevel,
+                inputParticles="particles_input",
+                outputParticles=particles_selected,
+            )
+        )
```

Comment:
Hi @paulgessinger,

Do you know what `ParticleSelector` does when no selections are specified? Naively I would have thought it did nothing, so why is this algorithm included here? It was there in the original `runFatras` example.

This is an issue because the seeding example doesn't call `ParticleSelector` before Fatras. If I add it (by calling `addFatras`), then I get a slightly different number of hits and `pytest` fails. That's why I added a flag to disable it, but it would be good to understand this oddity.

Alternatively, it might be more useful to allow more arbitrary selections specified by the caller. What do you think?

I hope this is a good place to put in random questions about this development, and @ you in my PR.

Thanks, Tim.

Reply:
Oh that's a good point. It's possible I skipped something when transcribing the original example to python. In principle the `ParticleSelector` should do nothing, but I could imagine due to numerical effects that it does throw out a few particles in edge cases. 

Did you by any chance have a look at the resulting written particles? Do they change noticeably?

Reply:
That's the odd thing: I can't see any difference between the initial and selected particles. The same number, values, and order. Fatras particles and hits are identical. Only when the seeding happens do things change: `track_finder_particles` has +/-1 or 2 tracks for some (23%) particles.
One other clue: even if I don't use the output, running `ParticleSelector` still changes the seeding. I guess this means there is something non-deterministic or state-dependent in the seeding algorithm - even single-threaded. Or just a bug in how we're using it. (I checked it's not just an output collection name clash.)
Could it be a problem with one of the mapping objects, if they use pointers that will be different if moved?

Reply:
Hm, that sounds weird indeed. So you don't see differences at the particle level, but they do change the seeds/tracks... 

If there's anything state-dependent, it also being present in single-threaded mode indicates that it's likely not a thread-safety issue. But other than that I don't know.

Reply:
Yes, that's what I see. I can't think of anything else I can test. Maybe make a mininal script to try to demonstrate the issue? Or see if it does the same with the example binaries? Neither a trivial amount of work, so let me know if it's worth it.